runtime/src/cost_model.rs

```diff
@@ -153,19 +160,25 @@ impl CostModel {
         data_bytes_cost
     }
 
-    fn get_transaction_cost(&self, transaction: &SanitizedTransaction) -> u64 {
-        let mut cost: u64 = 0;
+    fn get_transaction_cost(&self, transaction: &SanitizedTransaction) -> (u64, u64) {
+        let mut builtin_costs = 0u64;
+        let mut bpf_costs = 0u64;
 
         for (program_id, instruction) in transaction.message().program_instructions_iter() {
-            let instruction_cost = self.find_instruction_cost(program_id);
-            trace!(
-                "instruction {:?} has cost of {}",
-                instruction,
-                instruction_cost
-            );
-            cost = cost.saturating_add(instruction_cost);
+            // to keep the same behavior, look for builtin first
+            if let Some(builtin_cost) = BUILT_IN_INSTRUCTION_COSTS.get(program_id) {
```

Comment:
Could you explain why this separate branch is necessary?

It looks to me like `CostModel::initialize_cost_table()` inserts all the `BUILT_IN_INSTRUCTION_COSTS` into the the `instruction_execution_cost_table`, meaning that `self.find_instruction_cost()` would find them.

So to me, it looks like the main effect of this patch is to separate `builtin_costs` from `bpf_costs` in the metrics, which may not be worth the extra lookup.

Maybe it would be sufficient to call `initialize_cost_table()` in the tests, instead of working with a `CostModel::default()` instance?

Reply:
Yea, `BUILT_IN_INSTRUCTION_COSTS` used to mingled together with BPFs in same table, cause we didn't need to care the distinguish them. 

Now we need to. We could either add a flag to `instruction_execution_cost_table`, or let `cost_model` to encapsulate the details. I choose second one, by no longer insert `BUILT_IN_INSTRUCTION_COSTS` during `initialize_cost_table()`, so the `instruction_execution_cost_table` will be only bpfs that are added during runtime. By lookup builtin from its own hashmap instead of a combined map, we might saved a tiny bit of time?

Reply:
My main question is: Why do we need to distinguish builtins from bpf costs in the first place?